sopel/db.py

```diff
@@ -258,10 +304,37 @@ def execute(self, *args, **kwargs):
 
         The ``Result`` object returned is a wrapper around a ``Cursor`` object
         as specified by :pep:`249`.
+
+        .. deprecated:: 8.0
+
+            This method will be removed in Sopel 8.1, following the deprecation
+            of SQLAlchemy's :meth:`sqlalchemy.engine.Engine.execute`.
+
+            To perform raw SQL query, use the :class:`~SopelDB.engine`
+            attribute as per the migration guide from SQLAlchemy::
+
+                from sqlalchemy.sql import text
+
+                def my_command(bot, trigger):
+                    raw_sql = ' ... '  # your raw SQL
+                    # get a connection as a context manager
+                    with bot.db.engine.connect() as conn:
+                        res = conn.execute(text(raw_sql))
+                        data = res.fetchall()
```

Comment:
This works even if SQLAlchemy is being used on top of SQLite, right? I ask because SQLite doesn't really have "connections" like other RDBMS options do.

Reply:
It should, yes. And also, that's why I can't use the `future=True` on the engine/session, because the `execute` method exists. And that's why I deprecate it in Sopel 8 and I want to remove in 8.1. Because no one should use that!

Reply:
The only thing I have to say about this (thanks to the reminder I got when we chatted on IRC today) is that I'd rather only _remove_ things in a .0 version—in this case, wait until 9.0.

No objection to deprecating in 8.0, but deleting stuff in a "minor" release always rubs me the wrong way (even if other packages seem happy to throw things out right and left even in "patch" versions 😑).

Reply:
Alright, done in a future commit.